pya/helper/helpers.py

```diff
@@ -111,9 +110,21 @@ def buf_to_float(x, n_bytes=2, dtype=np.float32):
     return scale * np.frombuffer(x, fmt).astype(dtype)
 
 
+def _try_importing_pyaudio(fun_name):
+    try:
+        import pyaudio
+    except ImportError as e:
+        msg = (
+            f"Function '{fun_name}' requires pyaudio"
+        )
+        raise RuntimeError(msg) from e
+    else:
+        return pyaudio.PyAudio()
+
+
 def device_info():
     """Return a formatted string about available audio devices and their info"""
-    pa = pyaudio.PyAudio()
+    pa = _try_importing_pyaudio("device_info")
```

Comment:
as the return value is a `pyaudio.PyAudio()`, I suggest to change the function name to `_try_opening_pyaudio(fun_name)`